dot/network/notifications.go

```diff
@@ -156,15 +158,18 @@ func (s *Service) createNotificationsMessageHandler(info *notificationsProtocol,
 			// if we are the initiator and haven't received the handshake already, validate it
 			if hsData, has := info.getHandshakeData(peer); has && !hsData.validated {
 				logger.Trace("sender: validating handshake")
-				err := handshakeValidator(peer, hs)
+				err := handshakeValidator(peer, hs, stream)
 				if err != nil {
 					logger.Trace("failed to validate handshake", "protocol", info.protocolID, "peer", peer, "error", err)
 					hsData.validated = false
 					return errCannotValidateHandshake
 				}
 
-				hsData.validated = true
-				hsData.received = true
+				info.handshakeData.Store(peer, &handshakeData{
+					validated: true,
+					received:  true,
+					stream:    stream,
+				})
```

Comment:
since we are storing the stream in this function (which is good) we don't need to pass it to `handshakeValidator`